controllers/rule_controller.go

```diff
@@ -57,3 +116,71 @@ func (r *RuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&apiv1.ConfigMap{}).
 		Complete(r)
 }
+
+func (r *RuleReconciler) updateRulesConfigmap(ctx context.Context, data string) error {
+
+	var oathkeeperRulesConfigmap apiv1.ConfigMap
+
+	if err := r.Get(ctx, r.RuleConfigmap, &oathkeeperRulesConfigmap); err != nil {
+		if apierrs.IsNotFound(err) {
+
+			createMapFunc := func() error {
+
+				oathkeeperRulesConfigmap = apiv1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      r.RuleConfigmap.Name,
+						Namespace: r.RuleConfigmap.Namespace,
+					},
+					Data: map[string]string{dataKey: data},
+				}
+
+				return r.Create(ctx, &oathkeeperRulesConfigmap)
+			}
+
+			dieFunc := func() {
```

Comment:
Please extract outside scope of updateRulesConfigmap function

Reply:
done!